x-pack/platform/plugins/shared/inference/common/rest/inference_client.ts

```diff
@@ -23,6 +23,9 @@ export function createInferenceRestClient({
   const chatComplete = createChatCompleteRestApi({ fetch, signal });
 
   const client: InferenceClient = {
+    on: () => {
+      throw createInferenceInternalError(`on() is not supported on inference rest client`);
+    },
```

Comment:
As discussed, no straightforward way of doing it for the rest client, but I think it's fine.

Could silently do nothing instead of throwing, but I think throwing is better to at least identify if as some point we'll need to think about implementing it.